### PR TITLE
feat(student-dashboard): available tasks

### PR DESCRIPTION
**Pull request by nataliagulko**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [x] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

#1848 

#### 💡 Background and solution

- Show course schedule available tasks instead of course events
- Move table renderers (task, date) form Schedule component to Table 
- Rearrange cards order on Student Dashboard

MicrosoftTeams-image (59)

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/components/Table/renderers.tsx [hunk 1]
```diff
@@ -154,3 +156,46 @@ export const scoreRenderer = (item: CourseScheduleItemDto) => {
...
+
+    const isDeadlineSoon = now <= end && end.diff(now, 'hours') < 48 && !score;
+    const isCurrent = now >= start && now < end && !score;
+    const isDeadlineMissed = now >= end && end.diff(now, 'hours') >= -24 && !score;
+    const isPast = now > end || score;
+
+    if (isDeadlineSoon && date === 'end') color = 'warning';
+    else if (isCurrent && date === 'start') color = 'success';
+    else if (isDeadlineMissed && date === 'end') color = 'danger';
+    else if (isPast) color = 'secondary';
```
Comment (aaliakseyenka) [inline client/src/components/Table/renderers.tsx:185]: we do prefer brackets even for single line statements.
consider moving  the whole block to function with return statements

File: client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            rowKey="id"
+            size="middle"
+            columns={columns}
+          />
+        }
+      />
+      <style jsx>{`
+        :global(.ant-table-thead) {
+          display: none;
+        }
```
Comment (aaliakseyenka) [inline client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx:41]: that will be really global style which will affect the component on the same page

File: client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      `}</style>
+    </>
+  );
+}
```
Comment (aaliakseyenka) [inline client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx:45]: consider the case with empty tasks, seems there shouldn't be border 
Screenshot 2022-11-26 at 11 32 13

File: client/src/components/Table/renderers.tsx (hunk 1 above)
Comment (nataliagulko) [reply client/src/components/Table/renderers.tsx:185]: It was moved as-is from TableView renderers for reuse on Student Dashboard. I haven't done any change

File: client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx (hunk 2 above)
Comment (nataliagulko) [reply client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx:42]: Fixed, thank you!
It's not working without `:global`, so I've added class to table itself

File: client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx (hunk 3 above)
Comment (nataliagulko) [reply client/src/modules/StudentDashboard/components/NextEventCard/NextEventCard.tsx:49]: Fixed, thank you

**Review by aaliakseyenka — APPROVED** [review]

**Pull request merged**